src/dymaptic.GeoBlazor.Core/Scripts/jsBuilder.ts

```diff
@@ -227,7 +227,13 @@ export function buildJsPopupTemplate(popupTemplateObject: DotNetPopupTemplate, v
     if (viewId !== null) {
         let view = arcGisObjectRefs[viewId] as View;
         if (hasValue(view)) {
-            view.popup.on("trigger-action", async (event: PopupTriggerActionEvent) => {
+            if (hasValue(triggerActionHandler)) {
```

Comment:
do we want to start to convert more null checks over to 'hasValue'? I see you have included this multiple times here and above, but still have the null checks?

Reply:
It would be a good refactoring to do, to move to always using `hasValue`.